Sources/NIOFileSystem/Internal/SystemFileHandle.swift

```diff
@@ -1046,6 +1046,70 @@ extension SystemFileHandle: WritableFileHandleProtocol {
             try sendableView._resize(to: size).get()
         }
     }
+
+    public func setTimes(
+        lastAccessTime: FileInfo.Timespec?,
+        lastDataModificationTime: FileInfo.Timespec?
+    ) async throws {
+        try await self.threadPool.runIfActive { [sendableView] in
+            try sendableView._withUnsafeDescriptor { descriptor in
+                if lastAccessTime == nil, lastDataModificationTime == nil {
+                    // If the timespec array is nil, as per the `futimens` docs,
+                    // both the last accessed and last modification times
+                    // will be set to now.
+                    futimens(descriptor.rawValue, nil)
```

Comment:
Can you move this to `Syscalls.swift`? The pattern there is to wrap the syscall so we can more easily deal with platform differences and allows us to mock the sys calls for testing.

The function should then be wrapped up in a static function on `Syscall` returning a `Result<Void, Errno>`.